### PR TITLE
RFC: Update generate-at-breakpoints mixin

### PR DESCRIPTION
**Pull request by AlecRaeside**
Allow a mix of min and max media queries. eg 

```
@include generate-at-breakpoints('.u-position-fixed', (400 max, 401 min, desk max)) {
  position: fixed;
}
```

Change max-width media queries generated by the mixin from this format `.u-position-fixed-at-400` to `.u-position-fixed-up-to-400`

### CODE REVIEW

**Comment by kllevin** [comment]
LGTM

File: core/mixins/_generate-at-breakpoints.scss [hunk 1]
```diff
@@ -45,17 +49,29 @@ $mixin-breakpoint-name-placeholder: "{bp}";
...
+    @if type-of($breakpoint-name) == list {
+      $limit: nth($breakpoint-name, 2);
+      $breakpoint-name: nth($breakpoint-name, 1);
+    } @else {
+      // Palm is a special case where it uses a `max-width` media query
+      $limit: if($breakpoint-name == 'palm', 'max', 'min');
+    }
+
+    @if $limit == max {
+      $joiner: "up-to";
```
Comment (chris-pearce) [inline core/mixins/_generate-at-breakpoints.scss:72]: See my comments in Scally HipChat room about this:

screen shot 2014-10-08 at 8 27 24 am

@kllevin what do you think about this?

I think even if we don't change `min-width` mq's we definitely need to update the general mq `max-width` mixin so we're being consistent.

**Comment by chris-pearce** [comment]
@AlecRaeside can you update the intro comments to reflect these changes e.g. 

> > Generate classes which apply styling at different breakpoints which is fed from the breakpoints defined here: Core -> Settings -> Breakpoints **or any custom breakpoint**.

And update this section: https://github.com/AlecRaeside/scally/blob/generate-at-breakpoint-options/core/mixins/_generate-at-breakpoints.scss#L11-L15 to include the new `max-width` joiner and the `min-width` if we go with it.

File: core/mixins/_generate-at-breakpoints.scss (hunk 1 above)
Comment (AlecRaeside) [inline core/mixins/_generate-at-breakpoints.scss:72]: What do you mean by update the general max-width mq? The `respond-to` mixin doesn't generate classes

Comment (chris-pearce) [inline core/mixins/_generate-at-breakpoints.scss:72]: Of course! Sorry.

OK so what about changing the `min-width` joiner to `from` rather than `at`?

Comment (AlecRaeside) [inline core/mixins/_generate-at-breakpoints.scss:72]: Yea thats a good idea, Kate and I were discussing it earlier, some of the ideas were `-starting-at-` or `-starting-from-`. I like `-from-` as it is succinct. I'll update PR.

Comment (chris-pearce) [inline core/mixins/_generate-at-breakpoints.scss:72]: Yeah I prefer being more succinct.

**Comment by chris-pearce** [comment]
@AlecRaeside see my comment above about this bit:

> > Generate classes which apply styling at different breakpoints which is fed from the breakpoints defined here: Core -> Settings -> Breakpoints or any custom breakpoint.

**Comment by chris-pearce** [comment]
LGTM

**Comment by chris-pearce** [comment]
@kllevin FYI

**Pull request merged**